Source/engine/trn.cpp

```diff
@@ -65,4 +65,23 @@ std::optional<std::array<uint8_t, 256>> GetClassTRN(Player &player)
 	return std::nullopt;
 }
 
+std::optional<std::array<uint8_t, 256>> GetItemTRN(const Item &item)
+{
+	std::array<uint8_t, 256> trn;
+	const char *path;
+
+	switch (item._iCurs) {
+	case ICURS_BREAST_PLATE:
+		path = "data\\inv\\breast_plate.trn";
+		break;
+	default:
+		return std::nullopt;
+	}
+
+	if (LoadOptionalFileInMem(path, &trn[0], 256)) {

```

Comment:
It's not great for performance to talk to the filesystem on every frame.
Initialization should happen separately. The TRN should probably be part of the item data, rather than being in a switch here.

Reply:
I will add that to the tsv :)

Reply:
yeah if you can make this so that it's an option that can be added to the tsv data and we set this by default that means we won't have code specifically for this item and it allows modders to change the breast plate without having to fight this.

Reply:
I will expand this to also cover unique item tsv file, so trns can be applied there as well

Reply:
@glebm please help me. I'm trying to do it the right way but I'm struggling. I tried adding a `unique_ptr` to the `Item` struct to handle trns the same way monsters do, but I can use `unique_ptr` there because of how item data is copied I think? Any suggestions?

Reply:
If you blow up the image size further that yellow is going to stare at us even more.  The truth of the matter is that there is no indication that this is not really a fix and the yellow looks fine when the graphics are viewed at 100% size.